### DISCUSSION TITLE
Feedback

### DISCUSSION THREAD

**Discussion by VonHeikemen** [VonHeikemen/lsp-zero.nvim#1] [General]
Ask me anything about `lsp-zero`, whether you're using it or planning to use it. Feedback is also welcome, ideas on how to improve the defaults settings or the documentation,

**Comment by Lite5h4dow** [comment]
would be great if we could have something that allows us to pass configs conditionally to servers, insead of having to edit my files for every server i install and add a lsp.configure() for every server. prefereably it would take a callback that can accept the name of the language server and returns the same settings table configure does e.g. 

lsp-zero.bulk_configure(function(server) ... end)

in my usecase specifically it would end up looking like this

```lua
lsp-zero.bulk_configure(function(server)
  local config_exists, config = pcall(require, "lsp.settings." .. server.name)
  if config_exists then
    return config
  end
  return {}
end)
```
this way we can configure a way of initialising all servers quickly and easily and allowing newly installed servers to automatically start without having to edit the nvim script and add a configure for the new server

**Reply by Lite5h4dow** [reply]
im considering looking into making a pr to do this

**Comment by mreishus** [comment]
**Feedback**: Personally, I like "Enter" accepting autocompletion, but I find that sometimes it accepts autocompletes I wasn't expecting. I would prefer to have to press at least one key interacting with autocomplete (tab, down arrow, etc) before enter to accept works. 

In this example, I press `i` to go into insert mode, and type `test<CR>`, which accepts a snippet, but I just wanted to type the word `test` and go to the next line:

https://user-images.githubusercontent.com/937354/169557993-3550d96e-4191-47a7-882d-3b379da39bcf.mp4

In my vim-coc setup, I have `<CR>` to accept autocomplete, but I must select the first item before it is activated. If I don't, `<CR>` just goes to the next line. That way, autocomplete never "gets in the way" of my normal typing.

This is my first time typing vim-cmp, so I'm going to look through the docs to see if there is a way to customize this, but just wanted to share my feedback.

Btw, thank you for the plugin - I probably would have never tried Neovim LSP stuff without it. :+1:

**Reply by mreishus** [reply]
Found a way to do it!

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

local cmp = require('cmp')
local cmp_mappings = lsp.defaults.cmp_mappings()

cmp_mappings['<CR>'] = cmp.mapping.confirm({select = false})
lsp.setup_nvim_cmp({
  completion = {
    completeopt = 'menu,menuone,noinsert,noselect'
  },
  mapping = cmp_mappings,
})
lsp.setup()
```

**Reply by VonHeikemen** [reply]
Glad you found a solution.

The default behaviour works nicely with common snippets. I can autocomplete things like `if` statements and `for` loops without even thinking. Just `if<enter>` and the magic happens. This is something I really liked from sublime text's autocomplete.

**Comment by anoduck** [comment]
@VonHeikemen As I have come to expect from your work, quite excellent. As mentioned in an earlier discussion, I am now migrating from packer to lazy.nvim, and the addition of the package will provide the necessary means to complete the migration and satisfy current issues. It's discovery truly could not have been better timed. 

Thank you
